core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/toDataFrame.kt

```diff
@@ -187,6 +188,83 @@ public abstract class CreateDataFrameDsl<T> : TraversePropertiesDsl {
 
 // endregion
 
+// region toDataFrame overloads for built-in types
+
+/*
+Without overloads Iterable<String>.toDataFrame produces unexpected result
+
+
+```
+val string = listOf("aaa", "aa", null)
+string.toDataFrame()
+```
+=>
+  length
+0    3
+1    2
+2 null
+ */
+
+@JvmName("toDataFrameByte")
+public inline fun <reified B : Byte?> Iterable<B>.toDataFrame(): DataFrame<OfSingleValueColumn<B>> = toDataFrame {
+    OfSingleValueColumn<B>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameShort")
+public inline fun <reified S : Short?> Iterable<S>.toDataFrame(): DataFrame<OfSingleValueColumn<S>> = toDataFrame {
+    OfSingleValueColumn<S>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameInt")
+public inline fun <reified I : Int?> Iterable<I>.toDataFrame(): DataFrame<OfSingleValueColumn<I>> = toDataFrame {
+    OfSingleValueColumn<I>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameLong")
+public inline fun <reified L : Long?> Iterable<L>.toDataFrame(): DataFrame<OfSingleValueColumn<L>> = toDataFrame {
+    OfSingleValueColumn<L>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameString")
+public inline fun <reified S : String?> Iterable<S>.toDataFrame(): DataFrame<OfSingleValueColumn<S>> = toDataFrame {
+    OfSingleValueColumn<S>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameChar")
+public inline fun <reified C : Char?> Iterable<C>.toDataFrame(): DataFrame<OfSingleValueColumn<C>> = toDataFrame {
+    OfSingleValueColumn<C>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameBoolean")
+public inline fun <reified B : Boolean?> Iterable<B>.toDataFrame(): DataFrame<OfSingleValueColumn<B>> = toDataFrame {
+    OfSingleValueColumn<B>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameUByte")
+public inline fun <reified U : UByte?> Iterable<U>.toDataFrame(): DataFrame<OfSingleValueColumn<U>> = toDataFrame {
+    OfSingleValueColumn<U>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameUShort")
+public inline fun <reified U : UShort?> Iterable<U>.toDataFrame(): DataFrame<OfSingleValueColumn<U>> = toDataFrame {
+    OfSingleValueColumn<U>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameUInt")
+public inline fun <reified U : UInt?> Iterable<U>.toDataFrame(): DataFrame<OfSingleValueColumn<U>> = toDataFrame {
+    OfSingleValueColumn<U>::value from { it }
+}.cast()
+
+@JvmName("toDataFrameULong")
+public inline fun <reified U : ULong?> Iterable<U>.toDataFrame(): DataFrame<OfSingleValueColumn<U>> = toDataFrame {
+    OfSingleValueColumn<U>::value from { it }
+}.cast()
+
+@DataSchema
+public interface OfSingleValueColumn<T> {
```

Comment:
Bit of a strange name. I see you were going for "DataFrame of single value column", but hmm... I think "ValueColumn" would be also fine

Reply:
But we already have ValueColumn :c I wanted to avoid confusion with any "columns" that we have in API

Reply:
I see. Maybe `ValueProperty<T>`? Since we already have `KeyValueProperty<T>` as default `@DataSchema` interface.

Reply:
sounds good